sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,1157 @@
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to FEC mode on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+            fec_mode:
+                One of the following predefined constants, the actual FEC mode for the port to be configured:
+                     FEC_MODE_NONE,
+                     FEC_MODE_RS,
+                     FEC_MODE_FC
+        """
+
+        raise NotImplementedError
+
+    def set_anlt(self, enable, target):
+        """
+        This API enables/disables the cable auto-negotiation + link training (AN/LT).
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            enable:
+                a boolean, True if auto-negotiation + link training (AN/LT) is to be enabled
+                         , False if auto-negotiation + link training (AN/LT) is not to be enabled
+            target:
+                One of the following predefined constants, the actual target to get the stats on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+
+        Returns:
+            a boolean, True if the auto-negotiation + link training (AN/LT) enable/disable specified is configured
+                     , False if the auto-negotiation + link training (AN/LT) enable/disable specified is not configured
+        """
+
+        raise NotImplementedError
+
+    def get_anlt(self, target):
+        """
+        This API gets the auto-negotiation + link training (AN/LT) mode of the cable for corresponding port.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to get the AN/LT on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+            a boolean, True if auto-negotiation + link training (AN/LT) is enabled
+                     , False if auto-negotiation + link training (AN/LT) is not enabled
+        """
+
+        raise NotImplementedError
+
+    def get_event_log(self, clear_on_read=False):
+        """
+        This API returns the event log of the cable
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            clear_on_read:
+                a boolean, True if the log has to be cleared after read
+                         , False if the log is not to be cleared after read
+
+        Returns:
+           list:
+              a list of strings which correspond to the event logs of the cable
+        """
+
+        raise NotImplementedError
+
+    def get_pcs_stats(self, target):
```

Comment:
```suggestion
    def get_pcs_stats(self, target, clear_on_read=False):
```
Is it necessary to add clear_on_read to reset the pcs statistics?

Reply:
if the feature cannot be supported, you can return an error when the caller tries to enable clear_on_read= True;
I believe other vendors might be able to support this, hence it is kept as an argument.